### PR TITLE
Maint/quantity unit systems

### PR DESCRIPTION
**Pull request by acarvalh-work**
Fixes - https://github.com/ansys/pyfluent/pull/1664#issuecomment-1581523362

- Added a unit system class (and tests)
   - Users can create their own custom units and  unit systems and use them with the existing per-defined units and unit systems. 
- Added error handling for common errors (all modules)

### CODE REVIEW

**Review by hpohekar — APPROVED** [review]

File: src/ansys/fluent/core/quantity/dimensions.py [hunk 1]
```diff
@@ -120,3 +125,22 @@ def units(self):
...
+    @classmethod
+    def EXCESSIVE_PARAMETERS(cls):
+        return cls(
+            "Dimensions only accepts 1 of the following: (units) or (dimensions)."
+        )
+
+    @classmethod
+    def EXCESSIVE_DIMENSIONS(cls, len):
+        return cls(
+            f"`dimensions` must contain 9 values or less, currently there are {len}."
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/quantity/dimensions.py:145]: Shall we write **D**imensions in both cases

File: src/ansys/fluent/core/quantity/dimensions.py [hunk 2]
```diff
@@ -20,31 +22,32 @@ def __init__(
...
         self._units_table = q.UnitsTable()
-        unit_sys = unit_sys or "SI"
+        unit_sys = unit_sys or self._units_table.unit_systems["SI"]
 
         if units is not None:
             self._unit = units
             self._dimensions = self._units_to_dim(units=units)
 
         if dimensions:
+            if len(dimensions) > 9:
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/quantity/dimensions.py:35]: Let's expose this dimensions list size (=9) from this dimensions module. Then we can reference it here, in unit_system, and wherever else needed.

**Comment by seanpearsonuk** [comment]
@acarvalh-work looks very clean, thanks. I only made a couple of minor comments.

**Review by prmukherj — APPROVED** [review]

File: src/ansys/fluent/core/quantity/dimensions.py (hunk 1 above)
Comment (acarvalh-work) [reply src/ansys/fluent/core/quantity/dimensions.py]: Added more clarification to error messages, Dimensions was referring to the class while `dimensions` is the parameter that accepts a list of dimensions.

https://github.com/ansys/pyfluent/blob/16bfd1f51b57625263ffefce7cf960c6691ef305/src/ansys/fluent/core/quantity/dimensions.py#L144-L154

File: src/ansys/fluent/core/quantity/dimensions.py (hunk 2 above)
Comment (acarvalh-work) [reply src/ansys/fluent/core/quantity/dimensions.py]: Added static method to Dimensions to access max dimensions list length (9). exp -> `Dimensions.max_dim_len()`

https://github.com/ansys/pyfluent/blob/16bfd1f51b57625263ffefce7cf960c6691ef305/src/ansys/fluent/core/quantity/dimensions.py#L132-L135

File: src/ansys/fluent/core/quantity/dimensions.py (hunk 1 above)
Comment (seanpearsonuk) [reply src/ansys/fluent/core/quantity/dimensions.py]: thanks for clarifying

**Review by seanpearsonuk — APPROVED** [review]

**Pull request merged**